diff -ruNp Python-3.10.19.orig/configure.ac Python-3.10.19/configure.ac
--- Python-3.10.19.orig/configure.ac	2026-02-20 09:43:14.220768148 +0000
+++ Python-3.10.19/configure.ac	2026-02-20 09:45:00.548324143 +0000
@@ -746,6 +746,10 @@ AS_CASE([$ac_sys_system],
   [FreeBSD*], [MULTIARCH=""],
   [MULTIARCH=$($CC --print-multiarch 2>/dev/null)]
 )
+dnl Tweak MULTIARCH: GCC may report powerpc-linux-gnuspev1 but the
+dnl platform triplet uses powerpc-linux-gnuspe.
+AS_IF([test "x$MULTIARCH" = "xpowerpc-linux-gnuspev1"],
+      [MULTIARCH="powerpc-linux-gnuspe"])
 AC_SUBST([MULTIARCH])
 AC_MSG_RESULT([$MULTIARCH])
 
