diff --git a/drivers/net/ethernet/intel/igc/igc_main.c b/drivers/net/ethernet/intel/igc/igc_main.c
index 121f39b58a17..7a35f924a95a 100644
--- a/drivers/net/ethernet/intel/igc/igc_main.c
+++ b/drivers/net/ethernet/intel/igc/igc_main.c
@@ -3003,10 +3003,12 @@ static bool igc_clean_tx_irq(struct igc_q_vector *q_vector, int napi_budget)
 
 		switch (tx_buffer->type) {
 		case IGC_TX_BUFFER_TYPE_XSK:
+#if 0
 #if defined(CONFIG_TRACING)
 		/* Only use for RTCP KPI Measurement on Q2 */
 		if (tx_ring->queue_index == 2 && adapter->tstamp_config.tx_type == HWTSTAMP_TX_ON)
 			trace_printk("TX HW TS %lld\n", timestamp);
+#endif
 #endif
 			xsk_frames++;
 			break;
diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index a8299de4f7bd..afd579a28b31 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -2661,8 +2661,10 @@ static int stmmac_tx_clean(struct stmmac_priv *priv, int budget, u32 queue)
 				ktime_t tx_hwtstamp = { 0 };
 
 				stmmac_get_tx_hwtstamp(priv, p, &tx_hwtstamp);
+#if 0
 				trace_printk("XDP TX HW TS %llu\n",
 					     tx_hwtstamp);
+#endif
 			}
 		}
 
