Upstream-Status: Inappropriate

diff --git a/drivers/ntb/test/ntb_perf.c b/drivers/ntb/test/ntb_perf.c
index 618e0f8d7a62..6f900d05fa7a 100644
--- a/drivers/ntb/test/ntb_perf.c
+++ b/drivers/ntb/test/ntb_perf.c
@@ -76,20 +76,21 @@
 #include <linux/dmaengine.h>
 #include <linux/pci.h>
 #include <linux/ktime.h>
 #include <linux/slab.h>
 #include <linux/delay.h>
 #include <linux/sizes.h>
 #include <linux/workqueue.h>
 #include <linux/debugfs.h>
 #include <linux/random.h>
 #include <linux/ntb.h>
+#include <linux/string.h>
 
 #define DRIVER_NAME		"ntb_perf"
 #define DRIVER_VERSION		"2.0"
 
 MODULE_LICENSE("Dual BSD/GPL");
 MODULE_VERSION(DRIVER_VERSION);
 MODULE_AUTHOR("Dave Jiang <dave.jiang@intel.com>");
 MODULE_DESCRIPTION("PCIe NTB Performance Measurement Tool");
 
 #define MAX_THREADS_CNT		128
@@ -1057,29 +1058,29 @@ static int perf_copy_chunk(struct perf_thread *pthr,
 
 err_free_resource:
 
 	return ret;
 }
 
 
 static bool is_dmaengine_enabled(char* buffer, const char* str)
 {
     int bufferlen = strlen(buffer);
-    int strlen = strlen(str);
+    int strlen2 = strlen(str);
 
-    for (int i = 0; i <= bufferlen - strlen; i++) {
+    for (int i = 0; i <= bufferlen - strlen2; i++) {
         int j;
-        for (j = 0; j < strlen; j++) {
+        for (j = 0; j < strlen2; j++) {
             if (buffer[i + j] != str[j])
                 break;
         }
-        if (j == strlen) {
+        if (j == strlen2) {
             return true;
         }
     }
 
     return false;
 }
 
 static bool perf_dma_filter(struct dma_chan *chan, void *data)
 {
 	struct perf_ctx *perf = data;
