diff --git a/arch/powerpc/platforms/embedded6xx/Kconfig b/arch/powerpc/platforms/embedded6xx/Kconfig
index c6adff216fe6..541c284aeeda 100644
--- a/arch/powerpc/platforms/embedded6xx/Kconfig
+++ b/arch/powerpc/platforms/embedded6xx/Kconfig
@@ -1,14 +1,14 @@
 # SPDX-License-Identifier: GPL-2.0
 config EMBEDDED6xx
 	bool "Embedded 6xx/7xx/7xxx-based boards"
-	depends on PPC_BOOK3S_32 && BROKEN_ON_SMP
+	depends on PPC_BOOK3S_32
 
 config LINKSTATION
 	bool "Linkstation / Kurobox(HG) from Buffalo"
 	depends on EMBEDDED6xx
 	select MPIC
 	select FSL_SOC
 	select PPC_UDBG_16550 if SERIAL_8250
 	select DEFAULT_UIMAGE
 	imply MPC10X_BRIDGE if PCI
 	help
