Upstream-Status: Inappropriate

diff --git a/config/x86/meson.build b/config/x86/meson.build
index c3564b0011..7f45ded3dd 100644
--- a/config/x86/meson.build
+++ b/config/x86/meson.build
@@ -81,35 +81,36 @@ if (binutils_ok and cc.has_multi_arguments(cc_avx512_flags)
         cc_has_avx512 = true
         target_has_avx512 = (
                 cc.get_define('__AVX512F__', args: machine_args) != '' and
                 cc.get_define('__AVX512BW__', args: machine_args) != '' and
                 cc.get_define('__AVX512DQ__', args: machine_args) != '' and
                 cc.get_define('__AVX512VL__', args: machine_args) != ''
             )
     endif
 endif
 
-if not is_ms_compiler
-    # we require SSE4.2 for DPDK
-    if cc.get_define('__SSE4_2__', args: machine_args) == ''
-        message('SSE 4.2 not enabled by default, explicitly enabling')
-        machine_args += '-msse4'
-    endif
-endif
+#if not is_ms_compiler
+#    # we require SSE4.2 for DPDK
+#    if cc.get_define('__SSE4_2__', args: machine_args) == ''
+#        message('SSE 4.2 not enabled by default, explicitly enabling')
+#        machine_args += '-msse4'
+#    endif
+#endif
 
 # enable restricted transactional memory intrinsics
 # https://gcc.gnu.org/onlinedocs/gcc/x86-transactional-memory-intrinsics.html
 if cc.get_id() != 'msvc'
     machine_args += '-mrtm'
 endif
 
-base_flags = ['SSE', 'SSE2', 'SSE3','SSSE3', 'SSE4_1', 'SSE4_2']
+#base_flags = ['SSE', 'SSE2', 'SSE3','SSSE3', 'SSE4_1', 'SSE4_2']
+base_flags = ['SSE', 'SSE2', 'SSE3','SSSE3', 'SSE4_1']
 foreach f:base_flags
     compile_time_cpuflags += ['RTE_CPUFLAG_' + f]
 endforeach
 
 optional_flags = [
         'AES',
         'AVX',
         'AVX2',
         'AVX512BW',
         'AVX512CD',
diff --git a/lib/distributor/meson.build b/lib/distributor/meson.build
index 24988c5827..8de3a7af35 100644
--- a/lib/distributor/meson.build
+++ b/lib/distributor/meson.build
@@ -1,17 +1,13 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2017 Intel Corporation
 
 if is_windows
     build = false
     reason = 'not supported on Windows'
     subdir_done()
 endif
 
 sources = files('rte_distributor.c', 'rte_distributor_single.c')
-if arch_subdir == 'x86'
-    sources += files('rte_distributor_match_sse.c')
-else
-    sources += files('rte_distributor_match_generic.c')
-endif
+sources += files('rte_distributor_match_generic.c')
 headers = files('rte_distributor.h')
 deps += ['mbuf']
diff --git a/lib/meson.build b/lib/meson.build
index ce92cb5537..3ed51d76c9 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -56,21 +56,20 @@ libraries = [
         'reorder',
         'sched',
         'security',
         'stack',
         'vhost',
         'ipsec', # ipsec lib depends on net, crypto and security
         'pdcp', # pdcp lib depends on crypto and security
         'fib', #fib lib depends on rib
         'port', # pkt framework libs which use other libs from above
         'pdump', # pdump lib depends on bpf
-        'table',
         'pipeline',
         'graph',
         'node',
 ]
 
 always_enable = [
         'cmdline',
         'eal',
         'ethdev',
         'hash',
