From d10e87a93453ff9c14d5c7dda289a8134b503ae4 Mon Sep 17 00:00:00 2001
From: Alessio Igor Bogani <alessio.bogani@elettra.eu>
Date: Wed, 20 Jan 2016 16:26:25 +0100
Subject: [PATCH 2/3] Add device tree entry for rtc

Signed-off-by: Alessio Igor Bogani <alessio.bogani@elettra.eu>
---
 arch/powerpc/boot/dts/mvme5100.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/powerpc/boot/dts/mvme5100.dts b/arch/powerpc/boot/dts/mvme5100.dts
index a7eb6d25903d..7acede93f39a 100644
--- a/arch/powerpc/boot/dts/mvme5100.dts
+++ b/arch/powerpc/boot/dts/mvme5100.dts
@@ -81,6 +81,11 @@
 			interrupt-controller;
 			reg = <0xf3f80000 0x40000>;
 		};
+
+		rtc@80c8 {
+			compatible = "m48t59";
+			reg = <0x80c8 0x10>;
+		};
 	};
 
 	pci0: pci@feff0000 {
-- 
2.17.1

