From 2b2d1ced3df39c0b0c19f63d421b1be35870b595 Mon Sep 17 00:00:00 2001
From: Sudheesh Mavila <sudheesh.mavila@amd.com>
Date: Sun, 19 Jan 2025 20:16:20 +0530
Subject: [PATCH 12/38] fix to compliation error for ZEN1 ARCH

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
---
 arch/x86/events/amd/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/events/amd/core.c b/arch/x86/events/amd/core.c
index 8ed10366c4a2..4a8ac061b6d1 100644
--- a/arch/x86/events/amd/core.c
+++ b/arch/x86/events/amd/core.c
@@ -278,8 +278,8 @@ static u64 amd_pmu_event_map(int hw_event)
 	if (cpu_feature_enabled(X86_FEATURE_ZEN2) || boot_cpu_data.x86 >= 0x19)
 		return amd_zen2_perfmon_event_map[hw_event];
 
-	if (cpu_feature_enabled(X86_FEATURE_ZEN1))
-		return amd_zen1_perfmon_event_map[hw_event];
+	//if (cpu_feature_enabled(X86_FEATURE_ZEN1))
+	//	return amd_zen1_perfmon_event_map[hw_event];
 
 	return amd_perfmon_event_map[hw_event];
 }
-- 
2.25.1

