From 89de689aa88006e084fc83499358f2f98c26de5c Mon Sep 17 00:00:00 2001
From: Alessio Igor Bogani <alessio.bogani@elettra.eu>
Date: Tue, 7 Jun 2016 13:27:56 +0200
Subject: [PATCH 1/3] Enable L2 cache on all cores

---
 arch/powerpc/platforms/86xx/mvme7100.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/powerpc/platforms/86xx/mvme7100.c b/arch/powerpc/platforms/86xx/mvme7100.c
index ee983613570c..c1bed91587b1 100644
--- a/arch/powerpc/platforms/86xx/mvme7100.c
+++ b/arch/powerpc/platforms/86xx/mvme7100.c
@@ -33,6 +33,15 @@
 #define MVME7100_MAX6649_MASK		0x20
 #define MVME7100_ABORT_MASK		0x10
 
+#ifdef CONFIG_SMP
+static void __init smp_mvme7100_setup_cpu(int cpu_nr)
+{
+	mpic_setup_this_cpu();
+	_set_L2CR(_get_L2CR() | L2CR_L2E);
+}
+
+#endif
+
 /*
  * Setup the architecture
  */
@@ -47,6 +56,9 @@ static void __init mvme7100_setup_arch(void)
 
 #ifdef CONFIG_SMP
 	mpc86xx_smp_init();
+	smp_ops->setup_cpu = smp_mvme7100_setup_cpu;
+#else
+	_set_L2CR(_get_L2CR() | L2CR_L2E);
 #endif
 
 	fsl_pci_assign_primary();
-- 
2.17.1

