diff -ruNp libpdv-5.6.6.0.orig/pdv/dispatch/makefile libpdv-5.6.6.0/pdv/dispatch/makefile
--- libpdv-5.6.6.0.orig/pdv/dispatch/makefile	2023-04-07 12:39:24.495783371 +0000
+++ libpdv-5.6.6.0/pdv/dispatch/makefile	2023-04-07 12:39:40.355668097 +0000
@@ -1,6 +1,3 @@
-
-CCC=g++
-
 OPTIM=-O2
 #OPTIM=-g
 
@@ -15,7 +12,7 @@ INCLUDEFLAGS = -I. -I..
 
 CFLAGS += $(OPTIM) -fPIC $(INCLUDEFLAGS) $(OPTIONFLAGS)
 CXXFLAGS += $(OPTIM) $(INCLUDEFLAGS) $(OPTIONFLAGS)
-LDFLAGS += -L. -L.. -ledt -lpthread -lm -Wl,-rpath -Wl,$(PDVHOME)
+LDFLAGS += -L. -L.. -ledt -lpthread -lm
 
 DISPATCHLIB_STATIC=libdispatch.a
 DISPATCHLIB=libdispatch.so
@@ -34,8 +31,8 @@ all : $(DISPATCHLIB)
 $(DISPATCHLIBOBJS) : $(DISPATCHHEADERS)
 
 $(DISPATCHLIB) : $(DISPATCHLIBOBJS)
-	@echo $(CCC) -o $(DISPATCHLIB) -shared $(DISPATCHLIBOBJS) $(LOGOUT)
-	@$(CCC) -o $(DISPATCHLIB) -shared $(DISPATCHLIBOBJS) $(LOGOUT)
+	@echo $(CXX) -o $(DISPATCHLIB) -shared $(DISPATCHLIBOBJS) $(LOGOUT)
+	@$(CXX) -o $(DISPATCHLIB) -shared $(DISPATCHLIBOBJS) $(LOGOUT)
 	@$(SHOWOUT)
 	mv $(DISPATCHLIB) $(LIBDIR)
 
@@ -44,7 +41,7 @@ $(DISPATCHLIB_STATIC) : $(DISPATCHLIBOBJ
 	mv $(DISPATCHLIB_STATIC) $(LIBDIR)
 
 %.o:  %.cpp
-	$(CCC) -c $(CFLAGS) -o $@ $<
+	$(CXX) -c $(CFLAGS) -o $@ $<
 
 libs:   
 	cd ../edtimage; make
diff -ruNp libpdv-5.6.6.0.orig/pdv/edt_camlink/makefile libpdv-5.6.6.0/pdv/edt_camlink/makefile
--- libpdv-5.6.6.0.orig/pdv/edt_camlink/makefile	2023-04-07 12:39:24.495783371 +0000
+++ libpdv-5.6.6.0/pdv/edt_camlink/makefile	2023-04-07 12:39:40.355668097 +0000
@@ -1,6 +1,4 @@
 
-CCC=g++
-
 DEBFLAGS=-O2 -g -Wno-format-zero-length -Wno-deprecated-declarations -Wno-write-strings -Wno-ignored-qualifiers -Wno-unused-result
 #DEBFLAGS=-DDEBUG_CALLS -O2 -g
 
@@ -42,7 +40,7 @@ all : $(LIBCLSEREDT)  clser_example
 $(LIBCLSEREDTOBJS) : $(LIBCLSEREDTHEADERS)
 
 $(LIBCLSEREDT) : $(LIBCLSEREDTOBJS) $(LIBPDVOBJS)
-	$(CCC) -o $(LIBCLSEREDT) $(LDFLAGS) $(LIBCLSEREDTOBJS) $(LIBPDVOBJS) $(LOGOUT)
+	$(CXX) -o $(LIBCLSEREDT) $(LDFLAGS) $(LIBCLSEREDTOBJS) $(LIBPDVOBJS) $(LOGOUT)
 	@$(SHOWOUT)
 	mv $(LIBCLSEREDT) $(LIBDIR)
 
@@ -51,7 +49,7 @@ $(LIBCLSEREDT_STATIC) : $(LIBCLSEREDTOBJ
 	mv $(LIBCLSEREDT_STATIC) $(LIBDIR)
 
 %.o:  %.cpp
-	$(CCC) -c $(CFLAGS) -o $@ $<
+	$(CXX) -c $(CFLAGS) -o $@ $<
 
 fullclean: clean
 	-rm -f $(patsubst %,../%,$(TARGS)) 
@@ -69,7 +67,7 @@ list:
 
 
 clser_example: clser_example.o edt_camlink.h
-	$(CCC) clser_example.cpp -o clser_example  -L. -L.. -lclseredt
+	$(CXX) clser_example.cpp -o clser_example  -L. -L.. -lclseredt
 
 ALL_CFLAGS += $(DEBFLAGS) $(DO_MLOCK) $(BUILDOPTIONS)
 ALL_CFLAGS += -I.. -D_REENTRANT -fPIC
diff -ruNp libpdv-5.6.6.0.orig/pdv/edtimage/makefile libpdv-5.6.6.0/pdv/edtimage/makefile
--- libpdv-5.6.6.0.orig/pdv/edtimage/makefile	2023-04-07 12:39:24.495783371 +0000
+++ libpdv-5.6.6.0/pdv/edtimage/makefile	2023-04-07 12:39:40.355668097 +0000
@@ -1,15 +1,12 @@
-
 PDVHOME=..
 
-CCC = gcc
-
 OPTIM=-O2
 #OPTIM=-g
 
 OPTIONFLAGS = -Wno-format-zero-length -Wno-deprecated-declarations -Wno-write-strings -Wno-ignored-qualifiers -Wno-unused-result
 
-CFLAGS = -fPIC $(OPTIM) -I$(PDVHOME) $(OPTIONFLAGS)
-LDFLAGS = -L$(PDVHOME) -L. -lpdvplus -lpdv -lpthread -lm 
+CFLAGS += -fPIC $(OPTIM) -I$(PDVHOME) $(OPTIONFLAGS)
+LDFLAGS += -L$(PDVHOME) -L. -lpdvplus -lpdv -lpthread -lm 
 
 LIBDIR=..
 LIB= libedtimage.a
@@ -39,7 +36,7 @@ objclean:
 clean: objclean
 
 %.o:  %.cpp
-	$(CCC) -c $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 
 
diff -ruNp libpdv-5.6.6.0.orig/pdv/imgfiles/makefile libpdv-5.6.6.0/pdv/imgfiles/makefile
--- libpdv-5.6.6.0.orig/pdv/imgfiles/makefile	2023-04-07 12:39:24.495783371 +0000
+++ libpdv-5.6.6.0/pdv/imgfiles/makefile	2023-04-07 12:39:40.355668097 +0000
@@ -1,14 +1,12 @@
 PDVHOME=..
 
-CPP=g++
-
 OPTIM= -O2
 #OPTIM= -g
 
 OPTIONFLAGS = -Wno-unused-but-set-parameter -Wno-format-zero-length -Wno-deprecated-declarations -Wno-write-strings -Wno-ignored-qualifiers -Wno-unused-result
 
-CFLAGS= -fPIC $(OPTIM) -I$(PDVHOME) -I.. -I../jpeg-6b -I../tiff-4.0.3/libtiff -DUSE_JPEG $(OPTIONFLAGS)
-LDFLAGS= -L$(PDVHOME) -L. -lpdvplus -lpdv -lpthread -lm
+CFLAGS += -fPIC $(OPTIM) -I$(PDVHOME) -I.. -DUSE_JPEG $(OPTIONFLAGS)
+LDFLAGS += -L$(PDVHOME) -L. -lpdvplus -lpdv -lpthread -lm
 
 LIBDIR=..
 LIB= libimgfiles.a
@@ -26,7 +24,7 @@ $(LIBDIR)/libimgfiles.a : $(OBJS)
 	ar ur $(LIBDIR)/libimgfiles.a $(OBJS)
 
 %.o: %.cpp
-	$(CPP) -c $(CFLAGS) -o $@ $<
+	$(CXX) -c $(CFLAGS) -o $@ $<
 
 objclean:
 	-rm -fr *.o *.a core *~ ../$(LIB)
diff -ruNp libpdv-5.6.6.0.orig/pdv/makefile libpdv-5.6.6.0/pdv/makefile
--- libpdv-5.6.6.0.orig/pdv/makefile	2023-04-07 12:39:24.495783371 +0000
+++ libpdv-5.6.6.0/pdv/makefile	2023-04-07 12:40:02.447507349 +0000
@@ -4,15 +4,13 @@ SHELL=/bin/bash
 DEBUG = y
 
 # Allow specific variables to be exported to submakes.
-export LIBRARY JPEGLIB TIFFLIBS LIBTIFF
+export LIBRARY
 
 include includes.mk
 
 VER_SERIES=$(shell uname -r | cut -f1-2 -d.)
 VER_MINOR=$(shell uname -r | cut -f3 -d.)
 
-CC=gcc
-
 DEBFLAGS = -O2  -Wno-format-zero-length -Wno-deprecated-declarations -Wno-write-strings
 
 ALL_CFLAGS += $(DEBFLAGS) $(DO_MLOCK) $(BUILDOPTIONS)
@@ -49,51 +47,10 @@ ALL_TARGS = $(LIBRARY) $(PDVLIB) $(TARGE
 all: $(ALL_TARGS)
 
 #
-# included opensource tiff library tiff-4.0.3
-#
-LIBTIFF=libtiff.a
-TIFFDIR=tiff-4.0.3
-TIFFSRC=$(TIFFDIR)/libtiff
-TIFFLIBDIR=$(TIFFDIR)/libtiff/.libs
-TIFFLIBS=$(TIFFLIBDIR)/libtiff.a $(TIFFLIBDIR)/libtiff.so
-TIFFMAKE=$(TIFFDIR)/Makefile
-AM_DIR=$(shell automake --print-libdir)
-
-$(TIFFDIR):
-	unzip -o $(TIFFDIR).zip
-
-$(TIFFLIBS): $(TIFFDIR) $(TIFF) $(TIFFMAKE)
-	make -C $(TIFFDIR)
-
-$(LIBTIFF): $(TIFFLIBS)
-	cp -a $(TIFFLIBDIR)/libtiff.so* .
-
-ifneq ($(strip $(AM_DIR)),)
-CFG_GUESS=$(AM_DIR)/config.guess
-CFG_SUB=$(AM_DIR)/config.sub
-$(TIFFMAKE):
-	cd $(TIFFDIR) ; mv config/config.guess config.guess.bak ; ln -s $(CFG_GUESS) config/config.guess ; mv config/config.sub config.sub.bak ; ln -s $(CFG_SUB) config/config.sub ; ./configure --disable-jbig --disable-jpeg --disable-lzma --disable-pixarlog --disable-zlib
-else
-$(TIFFMAKE):
-	cd $(TIFFDIR) ; ./configure --disable-jbig --disable-jpeg --disable-lzma --disable-pixarlog --disable-zlib
-endif
-#
-# included opensource jpeg library jpeg-6b 
-#
-JPEGDIR=jpeg-6b
-JPEGLIB=libjpeg.a
-
-$(JPEGLIB):
-	make -C $(JPEGDIR) $(JPEGLIB)
-	cp $(JPEGDIR)/$(JPEGLIB) .
-
-#
 # these sources depend additionally on the tiff library 
 #
-gen_tiffs send_tiffs simple_clsend: LDFLAGS += -ltiff -L$(TIFFLIBDIR) -Wl,-rpath=$(TIFFLIBDIR)
+gen_tiffs send_tiffs simple_clsend: LDFLAGS += -ltiff
 gen_tiffs send_tiffs simple_clsend: LDLIBS += -ltiff
-gen_tiffs send_tiffs simple_clsend: CFLAGS += -I$(TIFFSRC)
-gen_tiffs send_tiffs simple_clsend: $(TIFFLIBS)
 
 # Conditionally add vlviewer make.
 -include vlviewer.mk
@@ -113,17 +70,11 @@ libobjclean:
 	-make -C edtimage objclean
 	-make -C imgfiles objclean
 
-tiffclean:
-	-make -C $(TIFFDIR) clean
-
-jpegclean:
-	-make -C $(JPEGDIR) clean
-
 ifndef EDTBUILD
-clean: objclean tiffclean jpegclean libobjclean
+clean: objclean libobjclean
 endif
 
-uninstall: objclean tiffclean jpegclean libobjclean unload
+uninstall: objclean libobjclean unload
 	-make -C module uninstall
 
 driver:
@@ -179,8 +130,8 @@ dispatch libdispatch.so:
 FRC:
 	
 $(PDVLIB) : $(LIBOBJS)
-	@echo $(CC) $(ALL_CFLAGS) -o $(PDVLIB) -shared $(LIBOBJS) $(LOGOUT)
-	@$(CC) $(ALL_CFLAGS) -o $(PDVLIB) -shared $(LIBOBJS) $(LOGOUT)
+	@echo $(CC) $(ALL_CFLAGS) -o $(PDVLIB) $(LDFLAGS) -shared $(LIBOBJS) $(LOGOUT)
+	@$(CC) $(ALL_CFLAGS) -o $(PDVLIB) $(LDFLAGS) -shared $(LIBOBJS) $(LOGOUT)
 	@$(SHOWOUT)
 
 $(STATICPDVLIB) : $(LIBOBJS)
diff -ruNp libpdv-5.6.6.0.orig/pdv/pciediag_src/makefile libpdv-5.6.6.0/pdv/pciediag_src/makefile
--- libpdv-5.6.6.0.orig/pdv/pciediag_src/makefile	2023-04-07 12:39:24.495783371 +0000
+++ libpdv-5.6.6.0/pdv/pciediag_src/makefile	2023-04-07 12:39:40.355668097 +0000
@@ -1,18 +1,11 @@
 include ../makefile.def
 
-CC=g++
-
 DEBFLAGS = -O2
 
 CFLAGS += $(DEBFLAGS) -I.. -I.
 LDFLAGS += -L..
 
-
-ifeq ($(EDTDEV),pdv)
-    LDLIBS += -lpdv
-else
-    LDLIBS += -ledt
-endif
+LDLIBS += -lpdv
 
 LDLIBS += -lpthread -lm -ldl 
 
@@ -21,11 +14,11 @@ OBJS = pciecfg.o pciediag.o
 all: pciediag objclean
 
 pciediag: $(OBJS) 
-	$(CC) $(CFLAGS) -o ../$@ $(OBJS) $(LDFLAGS) $(LDLIBS) 
+	$(CXX) $(CFLAGS) -o ../$@ $(OBJS) $(LDFLAGS) $(LDLIBS) 
 
 %.o : %.cpp
-	@echo $(CC) $(CFLAGS) 
-	$(CC) $(CFLAGS) -c $< 
+	@echo $(CXX) $(CFLAGS) 
+	$(CXX) $(CFLAGS) -c $< 
 
 objclean: 
 	rm -f *.o 
diff -ruNp libpdv-5.6.6.0.orig/pdv/pdvplus/makefile libpdv-5.6.6.0/pdv/pdvplus/makefile
--- libpdv-5.6.6.0.orig/pdv/pdvplus/makefile	2023-04-07 12:39:24.495783371 +0000
+++ libpdv-5.6.6.0/pdv/pdvplus/makefile	2023-04-07 12:39:40.355668097 +0000
@@ -1,10 +1,7 @@
-
-CCC = gcc
-
 DEBFLAGS = -O2  -Wno-format-zero-length -Wno-deprecated-declarations -Wno-write-strings -Wno-ignored-qualifiers -Wno-unused-result
 
-CFLAGS =  -fPIC -I.. $(DEBFLAGS)
-LDFLAGS = -L.. -L. -lpdvplus -lpdv -lpthread -lm
+CFLAGS +=  -fPIC -I.. $(DEBFLAGS)
+LDFLAGS += -L.. -L. -lpdvplus -lpdv -lpthread -lm
 
 LIB= libpdvplus.a
 LIBDIR= ..
@@ -26,7 +23,7 @@ objclean:
 clean: objclean
 
 %.o:  %.cpp
-	$(CCC) -c $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 
 
