diff -ruNp openssh-8.9p1.orig/sandbox-seccomp-filter.c openssh-8.9p1/sandbox-seccomp-filter.c
--- openssh-8.9p1.orig/sandbox-seccomp-filter.c	2022-02-23 12:31:11.000000000 +0100
+++ openssh-8.9p1/sandbox-seccomp-filter.c	2022-10-26 16:40:40.063995293 +0200
@@ -285,6 +285,9 @@ static const struct sock_filter preauth_
 #ifdef __NR_pselect6_time64
 	SC_ALLOW(__NR_pselect6_time64),
 #endif
+#ifdef __NR_ppoll_time64
+	SC_ALLOW(__NR_ppoll_time64),
+#endif
 #ifdef __NR_read
 	SC_ALLOW(__NR_read),
 #endif
