diff --git a/include/configs/socfpga_common.h b/include/configs/socfpga_common.h
index 2fc4a12841..a6bf3af81b 100644
--- a/include/configs/socfpga_common.h
+++ b/include/configs/socfpga_common.h
@@ -257,21 +257,25 @@ unsigned int cm_get_qspi_controller_clk_hz(void);
 
 #ifndef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"autoload=no\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
+	"bootfile=boot.scr.uimg\0" \
 	"bootm_size=0xa000000\0" \
-	"kernel_addr_r="__stringify(CONFIG_SYS_LOAD_ADDR)"\0" \
-	"fdt_addr_r=0x02000000\0" \
+	"loadaddr="__stringify(CONFIG_SYS_LOAD_ADDR)"\0" \
+	"fdtaddr=0x02000000\0" \
 	"scriptaddr=0x02100000\0" \
-	"scriptfile=u-boot.scr\0" \
-	"fatscript=if fatload mmc 0:1 ${scriptaddr} ${scriptfile};" \
-			"then source ${scriptaddr}; fi\0" \
-	"pxefile_addr_r=0x02200000\0" \
-	"ramdisk_addr_r=0x02300000\0" \
 	"socfpga_legacy_reset_compat=1\0" \
-	"prog_core=if load mmc 0:1 ${loadaddr} fit_spl_fpga.itb;" \
-		   "then fpga loadmk 0 ${loadaddr}:fpga-core-1; fi\0" \
+	"krnfile=zImage\0" \
 	SOCFPGA_BOOT_SETTINGS \
-	BOOTENV
+	"bootcmd=" \
+		"if ext4load mmc 0:3 ${loadaddr} /.osdir/default/lib/firmware/bitstream.rbf; " \
+		"then fpga load 0 ${loadaddr} ${filesize}; bridge enable; fi; " \
+		"run setconsole; for target in ${boot_targets}; do run loadscript_${target}; done; sleep 10; reset\0" \
+	"setconsole=setenv console ttyS0,${baudrate}n8\0" \
+	"boot_targets=sd\0" \
+	"loadscript_dhcp=dhcp; if tftp ${scriptaddr} ${bootfile}; then source ${scriptaddr}; run bootcmd_dhcp; fi;\0" \
+	"loadscript_sd=mmc rescan; if load mmc 0:1 ${scriptaddr} ${bootfile}; then source ${scriptaddr}; run bootcmd_sd; fi;\0" \
+	"loadscript_usb=usb start; if load usb 0:1 ${scriptaddr} ${bootfile}; then source ${scriptaddr}; run bootcmd_usb; fi;\0"
 
 #endif
 #endif
