diff -ruNp omniORB-4.3.2.orig/configure.ac omniORB-4.3.2/configure.ac
--- omniORB-4.3.2.orig/configure.ac	2025-01-22 12:12:44.326841696 +0100
+++ omniORB-4.3.2/configure.ac	2025-01-22 12:10:41.571519703 +0100
@@ -1,7 +1,7 @@
 dnl -*- M4 -*-
 AC_INIT([omniORB],[4.3.2],[bugs@omniorb-support.com])
 
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 AC_CONFIG_SRCDIR(bin/scripts/omkdirhier)
 AC_CONFIG_AUX_DIR(bin/scripts)
 
